src/components/CActorArgument.vue

```diff
@@ -0,0 +1,191 @@
+      "
+      :formatted="argType === 'date' || argType === 'datetime.date' ? 'll' : 'llll'"
+      :only-date="argType === 'date' || argType === 'datetime.date'"
+      :class="invalidInput ? ['invalid-input', 'border'] : ''"
+      @input="validity = $event !== null"
+    />
+    <div v-else-if="argType === 'list' || argType.match(/^typing.List/)" class="h-full">
+      <c-list-input
+        v-if="!child"
+        v-model="value"
+        :valueType="valueType"
+        :name="name"
+        @validity="validity = $event"
+      />
+      <input
+        v-else
+        type="text"
+        v-model="value"
+        :class="['arg-input', { 'invalid-input': invalidInput }]"
+      />
+    </div>
+    <div v-else-if="argType === 'dict' || argType.match(/^typing.Dict/)" class="h-full">
+      <c-dict-input
+        v-if="!child"
+        v-model="value"
+        :keyType="keyType"
+        :valueType="valueType"
+        :name="name"
+        @validity="validity = $event"
+      />
+      <input
+        v-else
+        type="text"
+        v-model="value"
+        :class="['arg-input', { 'invalid-input': invalidInput }]"
+      />
+    </div>
+    <div v-else class="flex h-full">
+      <input
+        type="text"
+        v-model="value"
+        :class="['arg-input', { 'invalid-input': invalidInput }]"
+      />
+      <div v-if="initialValue === undefined && !name.match(/_key$/)">
+        <label class="text-xs absolute px-1">Parse as :</label>
+        <select class="h-full pt-2" v-model="parseAs">
+          <option>String</option>
+          <option>Json</option>
+        </select>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import CDictInput from '@/components/CDictInput';
+import CListInput from '@/components/CListInput';
+import utils from '@/utils';
+
+export default {
+  name: 'CActorArgument',
+  props: {
+    name: String,
+    argType: String,
+    child: Boolean,
+    isInvalid: Boolean,
+    initialValue: undefined
+  },
+  components: {
+    CListInput,
+    CDictInput
+  },
+  data: function() {
+    let value = '';
+    let keyType = '';
+    let valueType = '';
+    let validity = true;
+    switch (this.argType) {
+      case 'int':
+      case 'float':
+        value = 0;
+        break;
+      case 'bool':
+        value = false;
+        break;
+      case 'date':
+      case 'datetime':
+        validity = false;
+        break;
+      default:
+        value = '';
+    }
+    if (this.argType.match('^typing.List')) {
+      valueType = this.argType.substring(this.argType.indexOf('[') + 1, this.argType.length - 1);
+    }
+    if (this.argType.match(/^typing.Dict/)) {
+      keyType = this.argType.substring(this.argType.indexOf('[') + 1, this.argType.indexOf(','));
+      valueType = this.argType.substring(this.argType.indexOf(',') + 2, this.argType.length - 1);
+    }
+    if (this.initialValue !== undefined) {
+      value = this.initialValue;
+    }
+    this.$emit('input', value);
```

Comment:
I will rename the validity event to validityUpdate. 'input' is a different case, because it is the event used to pass data with v-model. Not using it would mean removing all the v-models and adding more code to compensate. Is it ok if I leave it as "input' ?